test-case/check-suspend-resume.sh

```diff
@@ -132,13 +135,33 @@ main()
 # TODO: remove this after issue fixed.
     sleep 1
 
+    local keep_modules=true already_unloaded=false
+
+    if [ ${OPT_VAL['u']} = 1 ]; then
+        keep_modules=false
+    fi
+
+    sudo lsmod | grep -q snd.sof || {
+        already_unloaded=true
+        $keep_modules ||
+            dlogw 'modules already unloaded, ignoring option -u!'
+    }
+
+    $already_unloaded || $keep_modules || "$TOPDIR"/tools/kmod/sof_remove.sh ||
+        die "Failed to unload audio drivers"
+
     expected_wakeup_count=$(cat /sys/power/wakeup_count)
     expected_stats_success=$(cat /sys/power/suspend_stats/success)
     save_initial_stats
     for i in $(seq 1 $loop_count)
     do
         sleep_once "$i"
     done
+
```

Comment:
should there be a checkpoint here before reloading audio drivers?

Reply:
There could be but it's not really necessary because one of the last thing `sleep_once` did is to check the kernel logs and `die` if that failed. So if the execution makes it to this point it means the kernel logs are free of errors so far.

Reply:
I don't disagree, it's more that if you get an error while loading audio drivers it might be a bit difficult to check if the test worked until the point where you start loading driver. You'd get a fail and no intermediate information, that's it.

Reply:
You will see the difference because of the different `die` error message; one is `dump_and_die "Caught error in kernel log"` whereas the other one below is  `die "Found kernel error after reloading audio drivers"`

Setting the `$KERNEL_CHECKPOINT` is completely quiet so it won't help in that respect.

Reply:
Think of `$KERNEL_CHECKPOINT` as a way to avoid errors from previous tests but inside the same test looking for errors and dying is actually a "stronger" checkpoint.

Reply:
I will have to trust you on that, I don't have any context here.